### PR TITLE
Add support for optional durations

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: src/ser.rs [hunk 1]
```diff
@@ -32,3 +57,29 @@ impl Serialize for chrono::Duration {
...
+
+#[cfg(feature = "chrono")]
+impl Serialize for Option<chrono::Duration> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        struct Data<'a, V: 'a>(&'a V)
+        where
+            V: Serialize;
```
Comment (code-ape) [inline src/ser.rs:69]: I forget you can do this in a function. Nice addition. Definitely makes this read much cleaner 😄

File: src/ser.rs [hunk 2]
```diff
@@ -22,6 +22,31 @@ impl Serialize for std::time::Duration {
     }
 }
 
+impl Serialize for Option<std::time::Duration> {
```
Comment (code-ape) [inline src/ser.rs:25]: Probably a dump question, but why support both your own time type and `std::time:Duration` for `Serialize`?

File: src/de.rs [hunk 3]
```diff
@@ -21,6 +25,53 @@ impl<'de> Deserialize<'de> for std::time::Duration {
     }
 }
 
+impl<'de> Deserialize<'de> for Option<std::time::Duration> {
```
Comment (code-ape) [inline src/de.rs:28]: So, just adding the standard "looks like you could macro this if you wanted to cut code count" comment. Presumably not really worth it for 50 LoC and a single duplication.

**Review by code-ape — APPROVED** [review]
Learned something reviewing this! Thanks for adding me. Added a couple questions. Only other call out is that tests appear to cover deserialization but not serialization, if you care. (I'm also sleep deprived so forgive me if I totally missed it 😬) Regardless, approving because I think the code you wrote, was probably written that way on purpose.

File: src/ser.rs (hunk 2 above)
Comment (Jarema) [reply src/ser.rs:25]: AFAIK serde didn't have serialization of any `Duration` to nanoseconds, neither `chrono` nor `std::time` so it's nice to support more use cases 🙂 .

**Comment by Jarema** [comment]
LGTM!

File: src/de.rs (hunk 3 above)
Comment (Jarema) [reply src/de.rs:28]: I would stand by readibility > single duplication in this case 🙂 .

Comment (caspervonb) [reply src/de.rs:28]: Too little duplication to warrant a macro imo.

File: src/ser.rs (hunk 2 above)
Comment (caspervonb) [reply src/ser.rs:25]: Feature gated support for the chrono crate, it's a different type with a bit different representation and precision.

**Pull request merged**